recipes/net-snmp/all/conanfile.py

```diff
@@ -0,0 +1,213 @@
+import functools
+import os
+import stat
+from contextlib import contextmanager
+
+from conans import AutoToolsBuildEnvironment, ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+
+required_conan_version = ">=1.43.0"
+
+
+class NetSnmpConan(ConanFile):
+    name = "net-snmp"
+    description = (
+        "Simple Network Management Protocol (SNMP) is a widely used protocol "
+        "for monitoring the health and welfare of network equipment "
+        "(eg. routers), computer equipment and even devices like UPSs."
+    )
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "http://www.net-snmp.org/"
+    topics = ("snmp",)
+    license = "BSD-3-Clause"
+    settings = ("os", "arch", "compiler", "build_type")
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_ipv6": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_ipv6": True,
+    }
+    requires = ("openssl/1.1.1m",)
+    exports_sources = ("patches/*",)
+
+    @property
+    def _settings_build(self):
+        return getattr(self, "settings_build", self.settings)
+
+    @property
+    def _is_msvc(self):
+        return self._settings_build.compiler in ("Visual Studio", "msvc")
+
+    @property
+    def _is_debug(self):
+        return self.settings.build_type == "Debug"
+
+    def validate(self):
+        if self._settings_build.os == "Windows" and not self._is_msvc:
+            raise ConanInvalidConfiguration(
+                "net-snmp is setup to build only with MSVC on Windows"
+            )
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], strip_root=True)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def build_requirements(self):
+        if self._is_msvc:
+            self.build_requires("strawberryperl/5.30.0.1")
+
+    @contextmanager
+    def _add_write_permissions(self, file):
+        write = stat.S_IWRITE
+        saved_permissions = os.stat(file).st_mode
+        if saved_permissions & write == write:
+            yield
+            return
+        try:
+            os.chmod(file, saved_permissions | write)
+            yield
+        finally:
+            os.chmod(file, saved_permissions)
+
+    @contextmanager
+    def _replace_in_file(self, file):
```

Comment:
It's not clear why you need it instead of https://docs.conan.io/en/latest/reference/tools.html?#tools-replace-in-file

Reply:
I had a lot of things to replace in single files before making the PR and there is no facility in `tools` that either supports `with` or accepts a dict with from -> to changes. Just didn't think of removing it.

I recently learned that Homebrew accepts blocks for its `inreplace`. Missing functionality in Conan? 🤔

Reply:
Reduced replacements to multiple calls to `tools.replace_in_file` in https://github.com/conan-io/conan-center-index/pull/8668/commits/6e3f2a8c18ae90ffb510e12e43c6d43fe11fe5bf